pkg/controller/argocd/deployment_test.go

```diff
@@ -902,6 +902,30 @@ func TestReconcileArgoCD_reconcileServerDeploymentChangedToInsecure(t *testing.T
 	}
 }
 
+func TestReconcileArgoCD_reconcileRedisDeployment(t *testing.T) {
+	// tests reconciler hook for redis deployment
+	cr := makeTestArgoCD()
+	r := makeTestReconciler(t, cr)
+	Register(testReconcilerHook)
+	var expected int32 = 4
+
+	assert.NilError(t, r.reconcileRedisDeployment(cr))
+	d := &appsv1.Deployment{}
+	assert.NilError(t, r.client.Get(context.TODO(), types.NamespacedName{Name: cr.Name+"-redis", Namespace: cr.Namespace}, d))
+	assert.DeepEqual(t, &expected, d.Spec.Replicas)
+}
+
+func testReconcilerHook(cr *argoprojv1alpha1.ArgoCD, v interface{}) error {
+	switch o := v.(type) {
+	case *appsv1.Deployment:
+		var x int32 = 4
+		if o.ObjectMeta.Name == cr.ObjectMeta.Name+"-redis" {
+			o.Spec.Replicas = &x
+		}
+	}
+	return nil
+}
+
```

Comment:
You could reuse the `testDeploymentHook` function here instead of creating another function?

Reply:
`testDeploymentHook` would work for any deployment, i wanted to keep this test hook specifically for Redis Deployment.

Reply:
using `testDeploymentHook` as we don't need redis specific name change only need to verify the behaviour